gst/nnstreamer/tensor_source/tensor_src_iio.c

```diff
@@ -401,15 +733,230 @@ gst_tensor_src_iio_finalize (GObject * object)
+  gboolean ret = FALSE;
+
+  filename = g_strdup_printf ("%s/%s", base_dir, file);
+  if (!g_file_set_contents (filename, contents, strlen (contents), &error)) {
+    GST_ERROR ("Unable to write file %s with error %s\n", filename,
+        error->message);
+    g_error_free (error);
+    g_free (filename);
+    return ret;
+  } else {
+    ret = TRUE;
+  }
+
+  if (DBG) {
+    gchar *file_contents = NULL;
+    ret = FALSE;
+    if (!g_file_get_contents (filename, &file_contents, NULL, &error)) {
+      GST_ERROR ("Unable to read file %s with error %s\n", filename,
+          error->message);
+      g_error_free (error);
+    } else {
+      if (g_strcmp0 (contents, file_contents)) {
+        ret = TRUE;
+      }
+      g_free (file_contents);
+    }
+  }
+
+  g_free (filename);
+  return ret;
+}
+
+/**
+ * @brief write the int in to the file
+ */
+static gboolean
+gst_tensor_write_sysfs_int (GstTensorSrcIIO * self, const gchar * file,
+    const gchar * base_dir, const gint contents)
+{
+  gchar *contents_char = NULL;
+  gboolean ret;
+
+  contents_char = g_strdup_printf ("%d", contents);
+  ret = gst_tensor_write_sysfs_string (self, file, base_dir, contents_char);
+
+  g_free (contents_char);
+  return ret;
+}
+
+/**
+ * @brief set value to all the channels
+ */
+static gboolean
+gst_tensor_set_all_channels (GstTensorSrcIIO * self, const gint contents)
+{
+  GList *ch_list;
+  gchar *filename = NULL;
+  GstTensorSrcIIOChannelProperties *channel_prop = NULL;
+  gboolean ret = TRUE;
+
+  for (ch_list = self->channels; ch_list != NULL; ch_list = ch_list->next) {
+    channel_prop = (GstTensorSrcIIOChannelProperties *) ch_list->data;
+    filename = g_strdup_printf ("%s%s", channel_prop->name, EN_SUFFIX);
+    ret &=
+        gst_tensor_write_sysfs_int (self, filename, channel_prop->base_dir,
+        contents);
+    g_free (filename);
+  }
+
+  return ret;
+}
+
 /**
  * @brief start function, called when state changed null to ready.
+ * load the device and init the device resources
  */
 static gboolean
 gst_tensor_src_iio_start (GstBaseSrc * src)
 {
   /** load and init resources */
-  //FIXME: fill this function
+  GstTensorSrcIIO *self;
+  self = GST_TENSOR_SRC_IIO_CAST (src);
+  guint id;
+
+  // Find the device
+  id = gst_tensor_src_iio_get_id_by_name (IIO_BASE_DIR, self->device.name,
+      DEVICE_PREFIX);
+  if (G_UNLIKELY (id < 0)) {
+    GST_ERROR_OBJECT (self, "Cannot find the IIO device with name: %s\n",
+        self->device.name);
+    goto error_return;
+  }
+  self->device.id = id;
+  self->device.base_dir = g_strdup_printf ("%s%s%d/", IIO_BASE_DIR,
+      DEVICE_PREFIX, self->device.id);
+
+  // no more configuration for one shot mode
+  if (!g_strcmp0 (self->mode, MODE_ONE_SHOT)) {
+    goto safe_return;
+  }
+  // register the trigger
+  if (self->trigger.name != NULL) {
+    // verify if trigger is supported by our device
+    gchar *trigger_device_dir =
+        g_strdup_printf ("%s%s", self->device.base_dir, TRIGGER);
```

Comment:
Use Glib.Path for pathname operations.

Reply:
Updated here as well as at other places.